crates/glaredb/src/lib.rs

```diff
@@ -714,3 +724,76 @@ impl Operation {
         ))
     }
 }
+
+#[derive(Debug, Clone, Copy)]
+pub enum ClientType {
+    Cli,
+    Node,
+    Python,
+    Rust,
+}
+
+impl From<ClientType> for RemoteClientType {
+    fn from(ct: ClientType) -> Self {
+        match ct {
+            ClientType::Cli => RemoteClientType::Cli,
+            ClientType::Node => RemoteClientType::Node,
+            ClientType::Rust => RemoteClientType::Rust,
+            ClientType::Python => RemoteClientType::Python,
+        }
+    }
+}
+
+
+impl Default for ClientType {
+    fn default() -> Self {
+        Self::Rust
+    }
+}
+
+impl Display for ClientType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ClientType::Cli => write!(f, "cli"),
+            ClientType::Node => write!(f, "node"),
+            ClientType::Python => write!(f, "python"),
+            ClientType::Rust => write!(f, "rust"),
+        }
+    }
+}
+
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Arrow(#[from] ArrowError),
+    #[error(transparent)]
+    Anyhow(#[from] anyhow::Error),
+    #[error(transparent)]
+    DataFusion(#[from] DataFusionError),
+
+    #[error(transparent)]
+    Metastore(#[from] MetastoreError),
+    #[error(transparent)]
+    Exec(#[from] ExecError),
+    #[error(transparent)]
+    ConfigurationBuilder(#[from] ConnectOptionsBuilderError),
+
+    #[error("{0}")]
+    Other(String),
+}
```

Comment:
This should be named something other than `Error` since `Error` is standard rust trait.